File: Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& a) {
    int j=0,i=1;
    int n = a.size();
    if(n==0 || n==1) return;
    while(i<n){
        if(a[j]==0){
            if(a[i]!=0){
                swap(a[i],a[j]);
                i++;
                j++;
            }else{
                i++;
            }
        }else{
            i++;
            j++;
        }
    }
        
    }
};

File: Move_Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        // for(int i=0 ;i<nums.length ; i++){
        //     for(int j=0 ;j<nums.length-1 ; j++){
        //         if(nums[j]==0){
        //             nums[j] = nums[j+1];
        //             nums[j+1] = 0;
        //         }
        //     }
        // }
        int n = nums.length;
        int count=0;
        for(int i=count ; i<n ; i++){
            if(nums[i]!=0){
                nums[count]=nums[i];
                count++;
            }
        }
        for(int i=count ; i<n ; i++){
            nums[i]=0;
        }
    }
}